std/algorithm/searching.d

```diff
@@ -1784,35 +1788,54 @@ if (isRandomAccessRange!R1 && isBidirectionalRange!R2
     // auto needleBack = moveBack(needle);
     // Stage 1: find the step
     size_t step = 1;
-    auto needleBack = needle.back;
+    immutable needleBack = needle.back;
     needle.popBack();
     for (auto i = needle.save; !i.empty && i.back != needleBack;
          i.popBack(), ++step)
     {
     }
     // Stage 2: linear find
     size_t scout = needleLength - 1;
+outer:
     for (;;)
     {
         if (scout >= haystack.length)
         {
-            return haystack[haystack.length .. haystack.length];
+            return haystack[$ .. $];
```

Comment:
Escalated this [to the forum](http://forum.dlang.org/post/snwstlkqitqdblpovssc@forum.dlang.org).